<?xml version="1.0"?>
<launch>
  <arg name="config" default="true"/>
  <!-- <arg name="rviz_config" default="$(find baxter_moveit_config)/launch/moveit.rviz" /> -->
  <arg name="rviz_config" default="$(find RobotUtils)/launch/moveit_camera.rviz" />

  <!-- By default we do not overwrite the URDF. Change the following to true to change the default behavior -->
  <arg name="load_robot_description" default="false"/>
  <!-- Left and right electric gripper params. Set to true to check for collisions for their links -->
  <arg name="right_electric_gripper" default="true"/>
  <arg name="left_electric_gripper" default="true"/>
  <!-- Set the kinematic tips for the left_arm and right_arm move_groups -->
  <arg name="left_tip_name" default="left_gripper"/>
  <arg name="right_tip_name" default="right_gripper"/>
  <include file="$(find baxter_moveit_config)/launch/planning_context.launch">
    <arg name="load_robot_description" value="$(arg load_robot_description)"/>
    <arg name="left_electric_gripper" value="$(arg left_electric_gripper)"/>
    <arg name="right_electric_gripper" value="$(arg right_electric_gripper)"/>
    <arg name="left_tip_name" value="$(arg left_tip_name)"/>
    <arg name="right_tip_name" value="$(arg right_tip_name)"/>
  </include>

  <group ns="camera1">
    <!-- Moving further! -->
    <node pkg="tf" type="static_transform_publisher" name="camera_broadcaster" args="2 0 0.8 -0.56239588 -0.56239588  0.42861507  0.42861507 world camera1 10" />
    <node name="camera_info" pkg="rostopic" type="rostopic"
        args="pub camera_info sensor_msgs/CameraInfo
     '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: 'camera1'},
      height: 480, width: 640, distortion_model: 'plumb_bob',
      D: [0],
      K: [500.0, 0.0, 320, 0.0, 500.0, 240.0, 0.0, 0.0, 1.0],
      R: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0],
      P: [500.0, 0.0, 320, 0.0, 0.0, 500, 240, 0.0, 0.0, 0.0, 1.0, 0.0],
      binning_x: 0, binning_y: 0,
      roi: {x_offset: 0, y_offset: 0, height: 480, width: 640, do_rectify: false}}' -r 2"
      output="screen"/>
  </group>
  <group ns="camera2">
    <!-- Moving further! -->
    <node pkg="tf" type="static_transform_publisher" name="camera_broadcaster" args="2 0.4 0.8 -0.50328489 -0.61390888  0.47029135  0.38554669 world camera2 10" />
    <node name="camera_info" pkg="rostopic" type="rostopic"
        args="pub camera_info sensor_msgs/CameraInfo
     '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: 'camera2'},
      height: 480, width: 640, distortion_model: 'plumb_bob',
      D: [0],
      K: [500.0, 0.0, 320, 0.0, 500.0, 240.0, 0.0, 0.0, 1.0],
      R: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0],
      P: [500.0, 0.0, 320, 0.0, 0.0, 500, 240, 0.0, 0.0, 0.0, 1.0, 0.0],
      binning_x: 0, binning_y: 0,
      roi: {x_offset: 0, y_offset: 0, height: 480, width: 640, do_rectify: false}}' -r 2"
      output="screen"/>
  </group>

  <group ns="camera3">
    <node pkg="tf" type="static_transform_publisher" name="camera_broadcaster" args="2 -0.4 0.8 -0.61390888 -0.50328489  0.38554669  0.47029135 world camera3 10" />
    <node name="camera_info" pkg="rostopic" type="rostopic"
        args="pub camera_info sensor_msgs/CameraInfo
     '{header: {seq: 0, stamp: {secs: 0, nsecs: 0}, frame_id: 'camera3'},
      height: 480, width: 640, distortion_model: 'plumb_bob',
      D: [0],
      K: [500.0, 0.0, 320, 0.0, 500.0, 240.0, 0.0, 0.0, 1.0],
      R: [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0],
      P: [500.0, 0.0, 320, 0.0, 0.0, 500, 240, 0.0, 0.0, 0.0, 1.0, 0.0],
      binning_x: 0, binning_y: 0,
      roi: {x_offset: 0, y_offset: 0, height: 480, width: 640, do_rectify: false}}' -r 2"
      output="screen"/>
  </group>

  <arg name="kinect" default="false" />
  <arg name="xtion" default="false" />
  <arg name="camera_link_pose" default="0.15 0.075 0.5 0.0 0.7854 0.0"/>
  <include file="$(find baxter_moveit_config)/launch/move_group.launch">
    <arg name="kinect" value="$(arg kinect)" />
    <arg name="xtion" value="$(arg xtion)" />
    <arg name="camera_link_pose" default="$(arg camera_link_pose)"/>
    <arg name="allow_trajectory_execution" value="true"/>
  </include>

  <include file="$(find baxter_moveit_config)/launch/moveit_rviz.launch">
    <arg name="config" value="$(arg config)" />
    <arg name="rviz_config" value="$(arg rviz_config)" />
  </include>

  <include file="$(find baxter_moveit_config)/launch/default_warehouse_db.launch" />
</launch>
